Fix Android build

mcclure opened the pull request:
- Fix typos after "os" refactor
- Fix CMake finding javac

bjornbytes commented on CMakeLists.txt:
```diff
@@ -637,6 +637,7 @@ elseif(ANDROID)
   )
 
   if(LOVR_BUILD_EXE)
+    find_package(Java)
```
This already exists on line 59, which confuses me

mcclure replied:
Then I think this one is a merge error specific to my branch.

I will retest and remove this line.

bjornbytes commented on src/core/os_android.c:
```diff
@@ -280,27 +280,27 @@ void os_poll_events() {
 }
 
-void os_on_quit(quitCallback callback) {
+void os_on_quit(fn_quit callback) {
```
Also I think the function pointers here should have a star after them e.g. `fn_quit*`

mcclure replied:
You are right, as I wrote them these functions would crash if called

mcclure commented:
I tested this new version with lovr-lodr. It addresses the issues above. I still was not able to get it to run without the `-0 so` -> `-0 ''` change we discussed in Slack.

bjornbytes commented:
The `-0 so` thing can be tracked here https://github.com/nothings/stb/pull/960#discussion_r596584133

But I am surprised that `0bb556554dd59ca706290cfad4796f410631924c` didn't fix it

mcclure commented:
I didn't test with 0bb5565. I tested with 6eef950aa84 (the parent of this patch)

mcclure commented:
I pushed the wrong commit on Friday :( :(

Third try!! How's this?
